backend/ibutsu_server/controllers/dashboard_controller.py

```diff
@@ -88,3 +89,24 @@ def update_dashboard(id_, dashboard=None):
     session.add(dashboard)
     session.commit()
     return dashboard.to_dict()
+
+
+def delete_dashboard(id_):
+    """Deletes a dashboard
+
+    :param id: ID of the dashboard to delete
+    :type id: str
+
+    :rtype: tuple
+    """
+    dashboard = Dashboard.query.get(id_)
+    if not dashboard:
+        return "Not Found", 404
+    else:
+        widget_configs = WidgetConfig.query.filter(WidgetConfig.dashboard_id == dashboard.id).all()
+        for widget_config in widget_configs:
+            widget_config.dashboard_id = None
```

Comment:
Since by design a widget config should be aligned with a single dashboard, I do think we should delete the widget configs when the dashboard is deleted.

Reply:
I was worried about people potentially deleting widgets that they don't want to delete, but that's more of a UI issue that should be handled in the frontend.